tests/honey_badger.rs

```diff
@@ -133,16 +133,16 @@ where
     // Handle messages in random order until all nodes have output all transactions.
     while network.nodes.values_mut().any(node_busy) {
         // If a node is expecting input, take it from the queue. Otherwise handle a message.
-        let input_ids: Vec<_> = network
+        if let Some(id) = network
             .nodes
             .iter()
             .filter(|(_, node)| !node.instance().algo().has_input())
             .map(|(id, _)| *id)
-            .collect();
-        if let Some(id) = rng.choose(&input_ids) {
-            let queue = queues.get_mut(id).unwrap();
-            queue.remove_multiple(network.nodes[id].outputs().iter().flat_map(Batch::iter));
-            network.input(*id, queue.choose(&mut rng, 3, 10));
+            .choose(&mut rng)
```

Comment:
Same as above.

Reply:
But as above, it's just a test, and I don't think this is a performance bottleneck.